### PR TITLE
Algolia search

### PR DESCRIPTION
**Pull request by veronicacannon**
What's the best way to handle API keys in an open project?

### CODE REVIEW

File: app/views/members/_search.html.erb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<%= form_tag(members_path, method: :get) do %>
+  <%= text_field_tag :query, params[:query] %>
```
Comment (phlipper) [inline app/views/members/_search.html.erb]: The `params[:query]` opens you up to potential XSS here. Please consider using [`sanitize`](http://api.rubyonrails.org/classes/ActionView/Helpers/SanitizeHelper.html#method-i-sanitize) for the query value.

File: config/initializers/algoliasearch.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+AlgoliaSearch.configuration = {
+	application_id: "YE495E6TKG",
```
Comment (phlipper) [inline config/initializers/algoliasearch.rb]: Funky whitespace again here 👾 

Also a few things about these keys: 

* These values _(`application_id`, `api_key`)_ most likely shouldn't be hard-coded in this project anywhere
* You added placeholders in the `config/secrets.yml` for these, so you should use the built-in Rails secrets support here.

I'd recommend modifying the setup slightly to be something like the following:

In `config/secrets.yml`:

```ruby
algolia:
  api_key: <%= ENV.fetch("ALGOLIA_API_KEY") %>
  application_id: <%= ENV.fetch("ALGOLIA_APPLICATION_ID") %>
```

Note the `algolia` key to namespace the values. Also note the `ENV.fetch`, which will cause the application to crash on boot if the values are not available _(ensuring you don't have an environment inadvertently missing these values)_. To ease this in development, I recommend using something like [`dotenv`](https://github.com/bkeepers/dotenv) or `figaro` or similar and checking in a default `.env` file. The `.env` file doubles as documentation of what ENV values are necessary for the app to run. I like a pattern of defaulting the `.env` values to something like:

```
ALGOLIA_API_KEY=insecure-algolia_api_key
ALGOLIA_APPLICATION_ID=insecure-algolia_application_id
```

The `insecure-` prefix makes it obvious in development that these are meant to be sensitive values. The suffix matches the ENV variable name, which means that it is easy to know which value is missing if you are expecting something different. 

In the initializer here, you can then use the Rails secrets support as intended:

```ruby
AlgoliaSearch.configuration = {
  api_key: Rails.application.secrets.algolia["api_key"],
  application_id: Rails.application.secrets.algolia["application_id"],
  # ...
}
```

File: db/seeds.rb [hunk 3]
```diff
@@ -5,3 +5,46 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+
+
+Member.create(
+	username:  'suee',
```
Comment (phlipper) [inline db/seeds.rb]: 👾

Comment (phlipper) [inline db/seeds.rb]: You might consider adding Rubocop to the project to help keep changes like this consistent across the project and multiple committers.

File: app/controllers/members_controller.rb [hunk 4]
```diff
@@ -4,7 +4,14 @@ class MembersController < ApplicationController
   # GET /members
   # GET /members.json
   def index
-    @members = Member.all
+    @members = if params[:query]
+      # to do sanitize query 
+      # to do option for basic search
+      # Member.where('name LIKE ?', "%#{params[:query]}%")
+      Member.search(params[:query])
```
Comment (phlipper) [inline app/controllers/members_controller.rb]: A controller test for the conditional case here would be a nice addition.

File: config/initializers/algoliasearch.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+	api_key: "21c4fe4689e99b41c2ca180c74cbec6a",
```
Comment (phlipper) [inline config/initializers/algoliasearch.rb]: You might consider rolling these keys now that they have been shared in a public repo.

**Comment by veronicacannon** [comment]
Thank you @phlipper!  Greatly appreciate the detailed comments.

**Comment by veronicacannon** [comment]
Rebase the algolia search on top of the basic search branch.

**Pull request closed without merging**